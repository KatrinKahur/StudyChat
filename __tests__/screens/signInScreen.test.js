import SignInScreen from "../../src/screens/signInScreen";
import {fireEvent, render} from "@testing-library/react-native";
import {getAuth, signInWithEmailAndPassword} from "firebase/auth";
import '@testing-library/jest-native/extend-expect';
import renderer from "react-test-renderer";

beforeAll(() => {
    global.FAKE_EMAIL = "jest@studychat.dev"
    global.FAKE_PASSWORD = "FakePassword"
});

describe('test that error messages are printed', () => {
    it('test that SignInScreen renders correctly', () => {
        const { toJSON } = renderer.create(
            <SignInScreen />
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('not filling in the password will print an error message', () => {
        const { getByPlaceholderText, getByText, queryByTestId } = render(
            <SignInScreen />
        );

        fireEvent.changeText(
            getByPlaceholderText("Email.."),
            global.FAKE_EMAIL
        );
        fireEvent.press(getByText("Sign in"));
        expect(queryByTestId('message-container')).toHaveTextContent('A password is required');
    });

    it('not filling in the email field will print an error message', () => {
        const { getByPlaceholderText, getByText, queryByTestId } = render(
            <SignInScreen />
        );
        fireEvent.changeText(
            getByPlaceholderText("Password.."),
            global.FAKE_PASSWORD
        );
        fireEvent.press(getByText("Sign in"));
        expect(queryByTestId('message-container')).toHaveTextContent("An email is required");
    });

    it('not filling in the credentials will print an error message', () => {
        const { getByText, queryByTestId } = render(
            <SignInScreen />
        );
        fireEvent.press(getByText("Sign in"));
        expect(queryByTestId('message-container')).toHaveTextContent("Please fill in the credentials");
    });


    it('not existing user credentials will print an error message', async () => {
        const email = "some_email@email.dev";
        const password = "SomePassword";
        const { getByPlaceholderText, getByText, queryByTestId, findByText } = render(
            <SignInScreen />
        );
        fireEvent.changeText(
            getByPlaceholderText("Email.."),
            email
        );
        fireEvent.changeText(
            getByPlaceholderText("Password.."),
            password
        );
        fireEvent.press(getByText("Sign in"));
        expect(signInWithEmailAndPassword).toHaveBeenCalledWith(getAuth(), email, password);
        await findByText("User not found");
        expect(queryByTestId('message-container')).toHaveTextContent("User not found");

    });

    it('wrong password will print an error message', async () => {
        const { getByPlaceholderText, getByText, queryByTestId, findByText } = render(
            <SignInScreen />
        );
        const password = "SomePassword"
        fireEvent.changeText(
            getByPlaceholderText("Email.."),
            global.FAKE_EMAIL
        );
        fireEvent.changeText(
            getByPlaceholderText("Password.."),
            password
        );
        fireEvent.press(getByText("Sign in"));
        expect(signInWithEmailAndPassword).toHaveBeenCalledWith(getAuth(), global.FAKE_EMAIL, password);
        await findByText("Invalid password");
        expect(queryByTestId('message-container')).toHaveTextContent("Invalid password");
    })
})

test('pushing the signup button redirects the user to main screen', () => {
    const navigationProps = {
        navigation: {
            push: jest.fn()
        }
    }
    const { getByText } = render(
        <SignInScreen {...navigationProps}/>
    );
    fireEvent.press(getByText("Click here to create one!"));
    expect(navigationProps.navigation.push).toHaveBeenCalledWith("Sign-up");
})

test('successful sign in', () => {
    const { getByPlaceholderText, getByText } = render(
        <SignInScreen />
    );

    fireEvent.changeText(
        getByPlaceholderText("Email.."),
        global.FAKE_EMAIL
    );
    fireEvent.changeText(
        getByPlaceholderText("Password.."),
        global.FAKE_PASSWORD
    );
    fireEvent.press(getByText("Sign in"));
    expect(signInWithEmailAndPassword).toHaveBeenCalled();

})