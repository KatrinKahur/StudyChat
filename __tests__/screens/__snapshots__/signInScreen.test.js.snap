// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that error messages are printed test that SignInScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "bold",
          "marginBottom": 35,
          "textAlign": "center",
        }
      }
    >
      Welcome to StudyChat!
    </Text>
    <View
      style={
        Object {
          "minWidth": 0,
          "width": "55%",
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "marginTop": 5,
            "marginVertical": 10,
            "paddingBottom": 5,
            "paddingRight": 10,
            "paddingTop": 20,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Email.."
          rejectResponderTermination={true}
          style={
            Object {
              "fontSize": 17,
              "overflow": "hidden",
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingVertical": 3,
              "width": "100%",
            }
          }
          testID="emailInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "marginTop": 5,
            "marginVertical": 10,
            "paddingBottom": 5,
            "paddingRight": 10,
            "paddingTop": 20,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password.."
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "fontSize": 17,
              "overflow": "hidden",
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingVertical": 3,
              "width": "100%",
            }
          }
          testID="passwordInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "minWidth": 0,
          "width": "55%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1e90ff",
            "borderRadius": 5,
            "marginBottom": 15,
            "marginTop": 10,
            "paddingVertical": 17,
          }
        }
        testID="signInBtn"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 15,
            }
          }
        >
          Sign in
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginTop": 30,
          }
        }
      >
        No account yet? 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1e90ff",
            "borderRadius": 5,
            "marginBottom": 15,
            "marginTop": 10,
            "paddingVertical": 17,
          }
        }
        testID="signUpBtn"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 15,
            }
          }
        >
          Click here to create one!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
